libpius/path.py

```diff
@@ -0,0 +1,167 @@
+        # others.
+        if any(cmd.lower().endswith(ext.lower()) for ext in pathext):
+            files = [cmd]
+        else:
+            files = [cmd + ext for ext in pathext]
+    else:
+        # On other platforms you don't have things like PATHEXT to tell you
+        # what file suffixes are executable, so just pass on cmd as-is.
+        files = [cmd]
+
+    seen = set()
+    for dir in path:
+        normdir = os.path.normcase(dir)
+        if not normdir in seen:
+            seen.add(normdir)
+            for thefile in files:
+                name = os.path.join(dir, thefile)
+                if _access_check(name, mode):
+                    return name
+    return None
+#end of "Which" function
+
+def gpg_path():
+    if sys.platform == "win32":
+        gpg = which('gpg2')
+    elif sys.platform == "darwin":
+        #joining all possible paths for location of mac binaries
+        BIN_PATHS = MAC_BIN_PATHS+":"+LINUX_BIN_PATHS)
+        gpg = which('gpg2', path=BIN_PATHS)
+    elif sys.platform == "linux":
+        gpg = which('gpg2', path=LINUX_BIN_PATHS)
+    if gpg == "":
+        print("GPG2 could not be found! Do you have the correct permissions?")
+        sys.exit(1)
+    else:
+        return gpg
+
+def get_home():
+    if sys.platform == "win32":
+        return os.environ.get('APPDATA')
+    else:
+        return os.environ.get('HOME')
+
+def get_gpghome(usrhome):
+    if sys.platform == "win32":
+        return os.environ.get('GNUPGHOME', os.path.join(usrhome, 'gnupg'))
+    else:
+        return os.environ.get('GNUPGHOME', os.path.join(usrhome, '.gnupg'))
+
+def get_piushome(usrhome):
+    mode=os.F_OK | os.X_OK
+    if sys.platform == "win32":
+        return os.path.join(usrhome, 'pius')
+    else:
+        #check to see if xdg home is a user env variable
+        #set it if it is to piushome
+        if os.path.exists(os.path.join(usrhome, '.pius')) and os.access(os.path.join(usrhome,'.pius'), mode) and os.path.isdir(os.path.join(usrhome,'.pius')):
+            piushome = os.path.join(usrhome,'.pius')
+        elif os.path.exists(os.path.join(usrhome, '.piusrc')) and os.access(os.path.join(usrhome,'.piusrc'), mode) and os.path.isdir(os.path.join(usrhome,'.piusrc')):
+            piushome = os.path.join(usrhome,'.piusrc')
+        elif os.environ.get('XDG_CONFIG_HOME') != "":
+            piushome = os.path.join(os.environ.get('XDG_CONFIG_HOME'), 'pius')
+        #long test to see if the path exists, is able to be accessed, and if it is a dir.
+        #then join the usrhome path with ,config/.pius and return it to the program.
+        elif os.path.exists(os.path.join(usrhome, '.config')) and os.access(os.path.join(usrhome,'.config'), mode) and os.path.isdir(os.path.join(usrhome,'.config')):
+            piushome = os.path.join(usrhome,'.config/pius')
+        #fall back if above is not applicable
+        else:
+            piushome = os.path.join(usrhome, '.pius')
+        return piushome
```

Comment:
this method returns a dir in the if, a file in the first elif, a dir in the second, a dir in the third. and a file in the else